### ISSUE TITLE
Response#new raises OpenSSL::PKey::RSAError when encrypted response is badly malformed

### ISSUE THREAD

**Issue by xeger** [onelogin/ruby-saml#304]
Hello,

I've noticed that Response is leaking exceptions when I send it an encrypted response with a corrupted ciphertext. It seems that the design intent of the class is to gather errors while processing rather than raising exceptions.

If I submit a PR to rescue OpenSSL errors during decryption and translate them into errors to add to the errors-collection, will that be of interest to you?

**Comment by pitbulk** [comment]
Yes I think so, please include unit test on your PR.

**Comment by chrisbloom7** [comment]
Has there been any movement on this? @xeger, can you provide an example of rescuing this?

**Comment by xeger** [comment]
I looked into this and discovered that the gem has tests that _expect_ an OpenSSL raise, e.g.
https://github.com/onelogin/ruby-saml/blob/master/test/response_test.rb#L1262

Since there's at least one case of _expected_ behavior that raises OpenSSL::PKey::RSAError, I decided to rescue at the application level instead.

I can change the expectation if there is interest, i.e. rescue all OpenSSL PKey errors internally to Response and translate them into a validation error. I'm not sure whether this would break anyone (i.e. people who currently rescue the PKey error to distinguish between bad ciphertext/key and other validation errors).
